### ISSUE TITLE
rvm install ruby 2.4.2

### ISSUE THREAD

**Issue by camorazrushimoe** [rvm/rvm#4318]
I want to install ruby 2.4.2
But have problem, plz, help

I try:
`rvm install ruby-2.4.2`
or
`rvm install --disable-binary ruby-2.4.2`

result:
```
Checking requirements for osx.
Certificates in '/usr/local/etc/openssl/cert.pem' are already up to date.
Requirements installation successful.
Installing Ruby from source to: /Users/sergeyryabushko/.rvm/rubies/ruby-2.4.2, this may take a while depending on your cpu(s)...
ruby-2.4.2 - #downloading ruby-2.4.2, this may take a while depending on your connection...
curl: (3) <url> malformed
**There was an error(3).

No fallback URL can be calculated for /2.4/ruby-2.4.2.tar.bz2, open a ticket with full output here:

    https://github.com/wayneeseguin/rvm/issues


There has been an error fetching the ruby interpreter. Halting the installation.**
```

**Comment by mpapis** [comment]
could you run it with `rvm install ruby-2.4.2 --debug` and show us the output?

**Comment by RajRoR** [comment]
```
$ rvm install ruby-2.4.2 --debug 
Warning! PATH is not properly set up, $GEM_HOME is not set.
         Usually this is caused by shell initialization files. Search for PATH=... entries.
         You can also re-add RVM to your profile by running: rvm get stable --auto-dotfiles
         To fix it temporarily in this shell session run: rvm use ruby-2.2.5
         To ignore this error add rvm_silence_path_mismatch_check_flag=1 to your ~/.rvmrc file.
Warning, new version of rvm available '1.29.4', you are using older version '1.29.3'.
You can disable this warning with:    echo rvm_autoupdate_flag=0 >> ~/.rvmrc
You can enable  auto-update  with:    echo rvm_autoupdate_flag=2 >> ~/.rvmrc
ruby-2.4.2 - install
ruby-2.4.2 - #already removed src/ruby-2.4.2
ruby-2.4.2 - #already removed rubies/ruby-2.4.2
Searching for binary rubies, this might take some time.
Running(16): curl --fail --location --max-redirs 10 --max-time 1800 --silent --insecure --location --list-only --max-time 5 --head https://rvm_io.global.ssl.fastly.net/binaries/ubuntu/18.04/x86_64/ruby-2.4.2.tar.bz2 --insecure
Remote file does not exist https://rvm_io.global.ssl.fastly.net/binaries/ubuntu/18.04/x86_64/ruby-2.4.2.tar.bz2
Running(16): curl --fail --location --max-redirs 10 --max-time 1800 --silent --insecure --location --list-only --max-time 5 --head https://s3.amazonaws.com/jruby.org/downloads/ruby-2.4.2.tar.bz2 --insecure
Remote file does not exist https://s3.amazonaws.com/jruby.org/downloads/ruby-2.4.2.tar.bz2
Running(16): curl --fail --location --max-redirs 10 --max-time 1800 --silent --insecure --location --list-only --max-time 5 --head https://rubies.travis-ci.org/ubuntu/18.04/x86_64/ruby-2.4.2.tar.bz2 --insecure
Remote file does not exist https://rubies.travis-ci.org/ubuntu/18.04/x86_64/ruby-2.4.2.tar.bz2
rvm_remote_server_url3 not found
No remote file name found
Running(16): curl --fail --location --max-redirs 10 --max-time 1800 --silent --insecure --location --list-only --max-time 5 --head https://rvm_io.global.ssl.fastly.net/binaries/ubuntu/18.04/x86_64/ruby-2.4.2.tar.bz2 --insecure
Remote file does not exist https://rvm_io.global.ssl.fastly.net/binaries/ubuntu/18.04/x86_64/ruby-2.4.2.tar.bz2
Running(16): curl --fail --location --max-redirs 10 --max-time 1800 --silent --insecure --location --list-only --max-time 5 --head https://s3.amazonaws.com/jruby.org/downloads/ruby-2.4.2.tar.bz2 --insecure
Remote file does not exist https://s3.amazonaws.com/jruby.org/downloads/ruby-2.4.2.tar.bz2
Running(16): curl --fail --location --max-redirs 10 --max-time 1800 --silent --insecure --location --list-only --max-time 5 --head https://rubies.travis-ci.org/ubuntu/18.04/x86_64/ruby-2.4.2.tar.bz2 --insecure
Remote file does not exist https://rubies.travis-ci.org/ubuntu/18.04/x86_64/ruby-2.4.2.tar.bz2
rvm_remote_server_url3 not found
No remote file name found
No binary rubies available for: ubuntu/18.04/x86_64/ruby-2.4.2.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Free disk space 554218MB, required 440MB.
__rvm_setup_compile_environment_setup ruby-2.4.2
rvm_autolibs_flag=4
__rvm_setup_compile_environment_movable_early ruby-2.4.2
__rvm_setup_compile_environment_system_early ruby-2.4.2
__rvm_setup_compile_environment_requirements ruby-2.4.2
Checking requirements for ubuntu.
requirements code for ubuntu loaded
Found required packages: bash, curl, patch, bzip2, ca-certificates, gawk, g++, gcc, autoconf, automake, bison, ca-certificates, curl, libc6-dev, libffi-dev, libgdbm-dev, libncurses5-dev, libsqlite3-dev, libtool, libyaml-dev, make, openssl, patch, pkg-config, sqlite3, zlib1g, zlib1g-dev, libgmp-dev, libreadline-dev.
Installing requirements for ubuntu.
Log prefix: /home/rajkumar/.rvm/log/1533309769_ruby-2.4.2/
Log filesystem: ext4
Log max name length: 250
Log file: /home/rajkumar/.rvm/log/1533309769_ruby-2.4.2/update_system.log
[2018-08-03 20:52:49] requirements_debian_update_system
requirements_debian_update_system () 
{ 
    __rvm_try_sudo apt-get --quiet --yes update || { 
        \typeset __ret=$?;
        case ${__ret} in 
            100)
                rvm_error "There has been an error while updating your system using \`apt-get\`.
It seems that there are some 404 Not Found errors for repositories listed in:

    /etc/apt/sources.list
    /etc/apt/sources.list.d/*.list

Make sure that all repositories are available from your system and verify your setup by running manually:

    sudo apt-get update

Make sure that it works correctly before proceeding with RVM.

If you are working from the GUI instead of the terminal, you might want to verify and fix broken
repositories using \"Software & Updates\" application.
"
            ;;
        esac;
        return ${__ret}
    }
}
current path: /home/rajkumar/apps/pt/chute/keypool
PATH=/home/rajkumar/.local/bin:/home/rajkumar/.rvm/gems/ruby-2.2.5/bin:/home/rajkumar/.rvm/gems/ruby-2.2.5@global/bin:/home/rajkumar/.rvm/rubies/ruby-2.2.5/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/home/rajkumar/.rvm/bin
command(2): requirements_debian_update_system ruby-2.4.2
Updating system....There has been an error while updating your system using `apt-get`.
It seems that there are some 404 Not Found errors for repositories listed in:

    /etc/apt/sources.list
    /etc/apt/sources.list.d/*.list

Make sure that all repositories are available from your system and verify your setup by running manually:

    sudo apt-get update

Make sure that it works correctly before proceeding with RVM.

If you are working from the GUI instead of the terminal, you might want to verify and fix broken
repositories using "Software & Updates" application.

.....
Error running 'requirements_debian_update_system ruby-2.4.2',
please read /home/rajkumar/.rvm/log/1533309769_ruby-2.4.2/update_system.log
Requirements installation failed with status: 100.
__rvm_rm_rf already gone: /home/rajkumar/.rvm/tmp/8405*
```

**Comment by nodefiend** [comment]
@mpapis 

I am having trouble using RVM to install ruby 2.4.2 

Here is my output:


```
arning, new version of rvm available '1.29.7-next', you are using older version '1.29.3'.
You can disable this warning with:    echo rvm_autoupdate_flag=0 >> ~/.rvmrc
You can enable  auto-update  with:    echo rvm_autoupdate_flag=2 >> ~/.rvmrc
ruby-2.4.2 - install
ruby-2.4.2 - #already removed src/ruby-2.4.2
ruby-2.4.2 - #already removed rubies/ruby-2.4.2
Searching for binary rubies, this might take some time.
Running(16): curl --fail --location --max-redirs 10 --max-time 1800 --silent --insecure --location --list-only --max-time 5 --head https://rvm_io.global.ssl.fastly.net/binaries/osx/10.14/x86_64/ruby-2.4.2.tar.bz2 --insecure
Remote file does not exist https://rvm_io.global.ssl.fastly.net/binaries/osx/10.14/x86_64/ruby-2.4.2.tar.bz2
Running(16): curl --fail --location --max-redirs 10 --max-time 1800 --silent --insecure --location --list-only --max-time 5 --head https://s3.amazonaws.com/jruby.org/downloads/ruby-2.4.2.tar.bz2 --insecure
Remote file does not exist https://s3.amazonaws.com/jruby.org/downloads/ruby-2.4.2.tar.bz2
Travis binaries for OSX are not movable and can't be used outside of Travis environment. Skip that source.
No remote file name found
Running(16): curl --fail --location --max-redirs 10 --max-time 1800 --silent --insecure --location --list-only --max-time 5 --head https://rvm_io.global.ssl.fastly.net/binaries/osx/10.14/x86_64/ruby-2.4.2.tar.bz2 --insecure
Remote file does not exist https://rvm_io.global.ssl.fastly.net/binaries/osx/10.14/x86_64/ruby-2.4.2.tar.bz2
Running(16): curl --fail --location --max-redirs 10 --max-time 1800 --silent --insecure --location --list-only --max-time 5 --head https://s3.amazonaws.com/jruby.org/downloads/ruby-2.4.2.tar.bz2 --insecure
Remote file does not exist https://s3.amazonaws.com/jruby.org/downloads/ruby-2.4.2.tar.bz2
Travis binaries for OSX are not movable and can't be used outside of Travis environment. Skip that source.
No remote file name found
No binary rubies available for: osx/10.14/x86_64/ruby-2.4.2.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Free disk space 211941MB, required 440MB.
__rvm_setup_compile_environment_setup ruby-2.4.2
rvm_autolibs_flag=4
__rvm_setup_compile_environment_movable_early ruby-2.4.2
__rvm_setup_compile_environment_system_early ruby-2.4.2
__rvm_setup_compile_environment_requirements ruby-2.4.2
Checking requirements for osx.
requirements code for osx loaded
requirements lib type set to osx_brew
brew seems to be writable
Found required packages: autoconf, automake, libtool, pkg-config, coreutils, libyaml, readline, libksba, openssl@1.1.
Updating Homebrew...
==> Upgrading 1 outdated package, with result:
automake 1.15.1 -> 1.16.1
==> Upgrading automake 
Error: Permission denied @ unlink_internal - /usr/local/share/doc/automake
Requirements installation failed with status: 1.
__rvm_rm_rf already gone: /Users/sudo/.rvm/tmp/30716*

```

**Comment by pkuczynski** [comment]
First of all you should get newer version by running `rvm get master`. Secondly you have some issues with permissions. Run `brew doctor` to fix it.

**Comment by nodefiend** [comment]
@pkuczynski

**Comment by SarpongAbasimi** [comment]
There was an error(3).

No fallback URL can be calculated for /2.6/ruby-2.6.0.tar.bz2, open a ticket with full output here:

    https://github.com/rvm/rvm/issues


There has been an error fetching the ruby interpreter. Halting the installation.
chriss-MBP% rvm install ruby-2.6.0 --debug
Warning, new version of rvm available '1.29.8', you are using older version '1.29.7-next'.
You can disable this warning with:    echo rvm_autoupdate_flag=0 >> ~/.rvmrc
You can enable  auto-update  with:    echo rvm_autoupdate_flag=2 >> ~/.rvmrc
ruby-2.6.0 - install
ruby-2.6.0 - #already removed src/ruby-2.6.0
ruby-2.6.0 - #already removed rubies/ruby-2.6.0
Searching for binary rubies, this might take some time.
rvm_remote_server_url not found
No remote file name found
rvm_remote_server_url not found
No remote file name found
No binary rubies available for: osx/10.13/x86_64/ruby-2.6.0.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Free disk space 16817MB, required 440MB.
__rvm_setup_compile_environment_setup ruby-2.6.0
rvm_autolibs_flag=4
__rvm_setup_compile_environment_movable_early ruby-2.6.0
__rvm_setup_compile_environment_system_early ruby-2.6.0
__rvm_setup_compile_environment_requirements ruby-2.6.0
Checking requirements for osx.
requirements code for osx loaded
requirements lib type set to osx_brew
brew seems to be writable
Found required packages: autoconf, automake, libtool, pkg-config, coreutils, libyaml, libksba, zlib, openssl@1.1.
Installing requirements for osx.
Log prefix: /Users/chrissongz/.rvm/log/1561310544_ruby-2.6.0/
Log filesystem: local,
Log max name length: 250
Log file: /Users/chrissongz/.rvm/log/1561310544_ruby-2.6.0/update_system.log
[2019-06-23 18:22:24] requirements_osx_brew_update_system
requirements_osx_brew_update_system () {
	if __rvm_version_compare "${_system_version}" -ge 10.7
	then
		__rvm_detect_xcode_version_at_least 4.6.2 || __CLT_version_at_least 4.6.0 || {
			\typeset ret=$?
			rvm_error "
Xcode version older than 4.6.2 installed, download and install newer version from:

    http://connect.apple.com

After installation open Xcode, go to Downloads and install Command Line Tools.
"
			return $ret
		}
	fi
	brew update || {
		\typeset ret=$?
		rvm_error "Failed to update Homebrew, follow instructions at

    https://docs.brew.sh/Common-Issues

and make sure \`brew update\` works before continuing."
		return $ret
	}
}
current path: /Users/chrissongz
PATH=/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/Applications/Xamarin Workbooks.app/Contents/SharedSupport/path-bin:/Users/chrissongz/.rvm/bin
GEM_HOME=
GEM_PATH=
command(2): requirements_osx_brew_update_system ruby-2.6.0
Updating system - please wait
Log file: /Users/chrissongz/.rvm/log/1561310544_ruby-2.6.0/package_install_readline.log
[2019-06-23 18:22:26] requirements_osx_brew_libs_install
requirements_osx_brew_libs_install () {
	brew install "$@" --force || {
		\typeset ret=$?
		requirements_osx_brew_libs_error "installation"
		return $ret
	}
}
current path: /Users/chrissongz
PATH=/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/Applications/Xamarin Workbooks.app/Contents/SharedSupport/path-bin:/Users/chrissongz/.rvm/bin
GEM_HOME=
GEM_PATH=
command(2): requirements_osx_brew_libs_install readline
Installing required packages: readline - please wait
PATH=/usr/local/opt/coreutils/bin:/usr/local/opt/pkg-config/bin:/usr/local/opt/libtool/bin:/usr/local/opt/automake/bin:/usr/local/opt/autoconf/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/Applications/Xamarin Workbooks.app/Contents/SharedSupport/path-bin:/Users/chrissongz/.rvm/bin
rvm_configure_flags+=( --with-opt-dir=/usr/local/opt/libyaml )
rvm_configure_flags+=( --with-opt-dir=/usr/local/opt/libksba )
rvm_configure_flags+=( --with-opt-dir=/usr/local/opt/readline )
rvm_configure_flags+=( --with-opt-dir=/usr/local/opt/zlib )
rvm_configure_flags+=( --with-opt-dir=/usr/local/opt/openssl@1.1 )
Certificates bundle '/usr/local/etc/openssl@1.1/cert.pem' is already up to date.
Requirements installation successful.
__rvm_setup_compile_environment_movable ruby-2.6.0
__rvm_setup_compile_environment_parse_name ruby-2.6.0
__rvm_setup_compile_environment_system ruby-2.6.0
__rvm_setup_compile_environment_architectures ruby-2.6.0
rvm_architectures(0):.
__rvm_setup_compile_environment_gcc47plus ruby-2.6.0
__rvm_setup_compile_environment_bison ruby-2.6.0
__rvm_setup_compile_environment_flags_docs ruby-2.6.0
__rvm_setup_compile_environment_flags_shared_static ruby-2.6.0
__rvm_setup_compile_environment_flags_threads ruby-2.6.0
__rvm_setup_compile_environment_compatibility_flag ruby-2.6.0
found compiler: gcc
Installing Ruby from source to: /Users/chrissongz/.rvm/rubies/ruby-2.6.0, this may take a while depending on your cpu(s)...
ruby-2.6.0 - #downloading ruby-2.6.0, this may take a while depending on your connection...
/Users/chrissongz/.rvm/scripts/functions/support: line 170: cd: /Users/chrissongz/.rvm/archives: No such file or directory
Load checksums for /2.6/ruby-2.6.0.tar.bz2
    ...checksums not found
Fetch /2.6/ruby-2.6.0.tar.bz2 => ruby-2.6.0.tar.bz2
    ...no archive, downloading
Running(19): curl --fail --location --max-redirs 10 --max-time 1800 --connect-timeout 30 --retry-delay 2 --retry 3 --create-dirs -C - -o ruby-2.6.0.tar.bz2.part /2.6/ruby-2.6.0.tar.bz2
curl: (3) <url> malformed
There was an error(3).

No fallback URL can be calculated for /2.6/ruby-2.6.0.tar.bz2, open a ticket with full output here:

    https://github.com/rvm/rvm/issues


There has been an error fetching the ruby interpreter. Halting the installation.
__rvm_rm_rf already gone: /Users/chrissongz/.rvm/tmp/523*
chriss-MBP%
